### DISCUSSION TITLE
Как через функцию печати сформировать не однородную таблицу?

### DISCUSSION THREAD

**Discussion by Dreary01** [totumonline/totum-ru-issues-and-discussions#453] [Как это сделать?] [answered]
Коллеги, добрый день. Делаю генерацию смет для строителей. Есть форма (см скриншот). На сколько я понял текущая реализация позволяет сделать печатную форму только однородной таблицы?
Как быть, если мне надо сначала сформировать верх таблицы (заголовок), потом взять тип работ, далее в цикле перебрать эти работы, и в конце вывести сумму этих работ, и так повторить N Раз?
{заголовок таблицы} - поля, статично
{название этапа} - запустить на каждый этап
{работы} - перебор работ в конкретном этапе в цикле
{сумма} - сумма из цикла {работы}
![image](https://github.com/totumonline/totum-ru-issues-and-discussions/assets/8349892/0396bffb-bc56-4109-aa36-ac55012e117e)

**Comment by ttm-alexeypolunin** [comment, marked as answer]
Это через вложенный шаблоне делается:

```
example3: linkToPrint(template: "name"; data: $row3)
row3: rowCreate(field: "peremennaya1" = $row33)
row33: rowCreate(field: "template" = "inner_multi_template"; field: "data" = $row333)
row333: rowListCreate(field: "i_peremennaya1" = $listvalue1; field: "i_peremennaya2" = $listvalue2)
listvalue1: listcreate(item: "value1_row1"; item: "value1_row2")
listvalue2: listcreate(item: "value2_row1"; item: "value2_row2")
```

пример в доках паршивый, но другого нет

Смотрите там Пример: https://ru.docs.totum.online/print-and-csv#sx-example

**Reply by ttm-alexeypolunin** [reply]
Для большинства задач «в цикле перебрать» в тотум есть listReplace — перебирает список от начала до конца.

**Comment by Dreary01** [comment]
Этот пример я видел. Проблема в том, что у вас таблица однородная. А мне надо создать не однородную. Вот такую
![image](https://github.com/totumonline/totum-ru-issues-and-discussions/assets/8349892/b6602b56-278b-42c2-a8a0-2cab6c9fc471)
Получается что при обращении row33: rowCreate(field: "template" = "inner_multi_template"; field: "data" = $row333) я создаю строки повторяющиеся, и это ломает разметку строки с номерами Этапов и суммами промежуточных этапов

**Reply by ttm-alexeypolunin** [reply]
ну так может быть два вложенных шаблона — один в другой

тут 3 урровня:

1. Общий table
2. Этап - сумма
3. Работы в этапе

**Reply by erpic-ru** [reply]
[вот тут обсуждали](https://github.com/totumonline/totum-ru-issues-and-discussions/discussions/404)
